### PR TITLE
feat(delegates): add instant delegate withdrawal for a fee

### PR DESCRIPTION
**Pull request by vilenarios**

### CODE REVIEW

File: src/common/io.ts [hunk 1]
```diff
@@ -824,6 +824,39 @@ export class IOWriteable extends IOReadable implements AoIOWrite {
     params: {
       target: string;
       decreaseQty: number | mIOToken;
+      instant: boolean;
```
Comment (dtfiedler) [inline src/common/io.ts:827]: i would make this optional - and default to false
```suggestion
      instant?: boolean;
```

File: src/common/io.ts [hunk 2]
```diff
@@ -824,6 +824,39 @@ export class IOWriteable extends IOReadable implements AoIOWrite {
...
+    const { tags = [] } = options || {};
+    if (params.instant == true) {
+      return this.process.send({
+        signer: this.signer,
+        tags: [
+          ...tags,
+          { name: 'Action', value: 'Decrease-Delegate-Stake' },
+          { name: 'Target', value: params.target },
+          { name: 'Quantity', value: params.decreaseQty.valueOf().toString() },
+          { name: 'Instant', value: 'true' },
```
Comment (dtfiedler) [inline src/common/io.ts:840]:
```suggestion
          { name: 'Instant', value: instant || false },
```

Comment (dtfiedler) [reply src/common/io.ts:840]: look at the other tag combining to avoid this `if` block entirely

File: src/types/io.ts [hunk 3]
```diff
@@ -402,6 +402,14 @@ export interface AoIOWrite extends AoIORead {
     params: {
       target: WalletAddress;
       decreaseQty: number | mIOToken;
+      instant: true;
```
Comment (dtfiedler) [inline src/types/io.ts:405]:
```suggestion
      instan?t: boolean;
```

File: src/common/io.ts (hunk 1 above)
Comment (vilenarios) [reply src/common/io.ts:827]: Fixed 

https://github.com/ar-io/ar-io-sdk/commit/ae7be5caaa79f470eb9755c05c85467e98643d55

File: src/common/io.ts (hunk 2 above)
Comment (vilenarios) [reply src/common/io.ts]: How bout this? 

https://github.com/ar-io/ar-io-sdk/commit/5e82a6a02f2c19ef647b164b97140d0d7e04240e

File: src/common/io.ts [hunk 4]
```diff
@@ -824,17 +824,39 @@ export class IOWriteable extends IOReadable implements AoIOWrite {
...
     const { tags = [] } = options || {};
+
     return this.process.send({
       signer: this.signer,
       tags: [
         ...tags,
         { name: 'Action', value: 'Decrease-Delegate-Stake' },
         { name: 'Target', value: params.target },
         { name: 'Quantity', value: params.decreaseQty.valueOf().toString() },
+        { name: 'Instant', value: String(params.instant ?? false) },
```
Comment (dtfiedler) [inline src/common/io.ts:840]:
```suggestion
        { name: 'Instant', value: params.instant ? "true" : "false" },
```

**Review by dtfiedler — APPROVED** [review]

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**